### ISSUE TITLE
Implement a freebusy interface

### ISSUE THREAD

**Issue by gjanssens** [owncloud/core#25169]
Freebusy information allows calendar apps to more intelligently propose dates for events based on the availability of the selected participants.

It's like sharing your calendar with others, however it will only show when you are busy, not the details for each event.

It would be great if the owncloud calendar app would have an interface to publish this information based on the events data it has.

I'm currently trying to query this information from kontact/korganizer.
From searching a bit through the owncloud code for version 9.0.1 it looks like sabredav has some support for it, but I didn't find whether the code also uses it and does publish freebusy information. If it does, I am only missing the documentation on how to use it. If not it would be great to have this implemented.

Thank you.

**Comment by DeepDiver1975** [comment]
@georgehrke can you comment on this one? THX

**Comment by georgehrke** [comment]
FreeBusy Information can be provided using the Schedule plugin - see http://sabre.io/dav/scheduling/

**Comment by georgehrke** [comment]
tldr: CalDAV Backend must implement `Sabre\CalDAV\Backend\SchedulingSupport`

**Comment by DeepDiver1975** [comment]
> tldr: CalDAV Backend must implement Sabre\CalDAV\Backend\SchedulingSupport

:heavy_check_mark:

**Comment by DeepDiver1975** [comment]
> FreeBusy Information can be provided using the Schedule plugin - see http://sabre.io/dav/scheduling/

:heavy_check_mark: remote.php/dav does load the scheduling plugin
:negative_squared_cross_mark: remote.php/caldav does NOT load the plugin

**Comment by herrdeh** [comment]
As I dont have control over my owncloud server, I'd be grateful if it could be implemented into the standard release.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).

**Comment by pnull** [comment]
Is this really closed? Or do you see any chance that we can get that feature anytime soon?

**Comment by herrdeh** [comment]
I still would like this feature very much !

**Comment by DeepDiver1975** [comment]
Both endpoints (remote.php/caldav and remote.php/dav) are supporting scheduling (which includes freebusy) for quite some time now.

Which owncloud versions are you using? 
Which endpoint are you connected to?
How are you trying to retrieve freebusy information?

THX

**Comment by herrdeh** [comment]
Hello, great to hear. I tried to google "owncloud free busy", but found only outdated info. Could you give me a hint to find info how to deal with this feature?

I'm using evolution PIM on linux and acalendar on android - and the web interface of course. 
OC 10.0.10

Cheers,
Wolf

**Comment by DeepDiver1975** [comment]
I am not top deep into freebusy feature. According to the Docs the schedule Plugin has to be added. Which is done Long Time ago.

**Comment by herrdeh** [comment]
Hm. There are some still open issues on that, 582 and 696 for example.
I still cannot find and docu on it, and I cannot find a "schedule" plugin in the market. 
The whole thing seem to be a little ghostish. 

Would somebody with enough insight please be so kind and push me in the right direction - how can I enable and use the freebusy feature, if it exists... - There must be some documentation on that, but all I can google is are my own requests on it...

Cheers,
Wolf

**Comment by DeepDiver1975** [comment]
As gar as i can tell it is enabled and should work out of the Box. 
Lets See If i find the Time to have a closer look.

**Comment by herrdeh** [comment]
Am Montag, den 17.06.2019, 14:22 -0700 schrieb Thomas Müller:
> As gar as i can tell it is enabled and should work out of the Box.
> Lets See If i find the Time to have a closer look.

I don't say it does not work. I just say that I have no idea how to use
it - and it seems some other people do so as well.

Thanks

**Comment by herrdeh** [comment]
And I wonder how I could configure it....  At least, I should...
- choose which Calendars are observed and which not
- choose how much time forward and backward in time is published
- which days of the week and which period of time within the day is published
and probably some more controls....

**Comment by DeepDiver1975** [comment]
As far as I can tell from my local research evolution is not using the free-busy mechanisms as implemented per https://tools.ietf.org/html/rfc4791#section-7.10

Evolution work with Exchange and Novel GroupWare or you can setup a Free/Busy Url within the contact.
The later is out of scope for SabreDav and would need some special implementation within ownCloud
see https://help.gnome.org/users/evolution/stable/calendar-free-busy.html.en

Other CalDAV clients need to be analysed accordingly .....

**Comment by herrdeh** [comment]
We seem to have a misunderstanding. 
The OP (even the title of the thread) says "implement a freebusy interface" - and that was my intention as well. To make freebusy permanently available, I think it has to be a service of the server, not the client. And it needs an interface (at the owncloud server), as I pointed out two postings above.
